docs/operators/regexparser.md

```diff
@@ -0,0 +1,266 @@
+Regex Parser Operator
+=====================
+
+## Operator Objective
+RegexParser is designed to parse records based on a regex pattern and construct a concrete java class also known as ["POJO"](https://en.wikipedia.org/wiki/Plain_Old_Java_Object) out of it. User needs to provide the regex pattern and schema definition to describe the data pattern. Based on regex pattern, the operator will split the data and then schema definition will be used to map the incoming record to POJO. User can also provide date format if any, in the schema. The supported constraints are listed in [constraints table](#constraints). 
+
+The regex pattern has to match the tuple in its entirety. Valid records will be emitted as POJOs while invalid ones are emitted on the error port with an error message.
+
+**Note**: field names of POJO must match field names in schema and in the same order as it appears in the incoming data.
+
+## Overview
+The operator is **idempotent**, **fault-tolerant** and **partitionable**.
+
+## Operator Information
+1. Operator location: **_malhar-contrib_**
+2. Available since: **_3.7.0_**
+3. Operator state: **_Evolving_**
+4. Java Package: [com.datatorrent.contrib.parser.RegexParser](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/RegexParser.java)
+
+
+## <a name="props"></a>Properties of RegexParser
+User needs to set the `schema` JSON string which describes data as well as specify the format on date fields if any.
```

Comment:
Add a note here:
In the examples below _{ApplicationName}_ and _{OperatorName}_ are placeholders for the respective names of the application and the operator.